src/NetMQ/Core/Transports/ByteArraySegment.cs

```diff
@@ -57,7 +57,7 @@ public ByteArraySegment(ByteArraySegment otherSegment, int offset)
         /// <summary>
         /// Get the number of bytes within the buffer that is past the Offset (ie, buffer-length minus offset).
         /// </summary>
-        public int Size => m_innerBuffer.Length - Offset;
+        public int Size => m_innerBuffer == null ? 0 : m_innerBuffer.Length - Offset;
```

Comment:
I'm not sure about this change. Code shouldn't be using an initialised version of this struct. The buffer should always be non-null. If it's null here, then there's a bug somewhere else. That is the bug that should probably be identified and fixed. At least that's my working assumption.